lib/widgets/lightbox.dart

```diff
@@ -492,11 +493,25 @@ class _VideoLightboxPageState extends State<VideoLightboxPage> with PerAccountSt
     _controller?.removeListener(_handleVideoControllerUpdate);
     _controller?.dispose();
     _controller = null;
+    WakelockPlus.disable();
     super.dispose();
   }
 
-  void _handleVideoControllerUpdate() {
-    setState(() {});
+  Future<void> _handleVideoControllerUpdate() async {
+    if (mounted) setState(() {});
+
+    final wakelockEnabled = await WakelockPlus.enabled;
+    if (_controller!.value.isPlaying) {
+      if (!wakelockEnabled) {
+        assert(debugLog('Wakelock enabled'));
+        await WakelockPlus.enable();
+      }
+    } else {
+      if (wakelockEnabled) {
+        assert(debugLog('Wakelock disabled'));
+        await WakelockPlus.disable();
+      }
+    }
```

Comment:
It seems unfortunate to have an async gap here, with `await WakelockPlus.enabled;`. Then we have to think about bugs that could come from race conditions.

I see we're using the resolved value in order to avoid making redundant calls to `WakelockPlus.{enable,disable}`. But in the library's docs, it says that's not necessary:

https://pub.dev/documentation/wakelock_plus/latest/wakelock_plus/WakelockPlus-class.html
> You do not need to worry about making redundant calls, e.g. calling [WakelockPlus.enable](https://pub.dev/documentation/wakelock_plus/latest/wakelock_plus/WakelockPlus/enable.html) when the wakelock is already enabled as the plugin handles this for you, i.e. it checks the status to determine if the wakelock is already enabled or disabled.

So could the implementation really just be:

```dart
  Future<void> _handleVideoControllerUpdate() async {
    setState(() {});
    if (_controller == null) return;
    
    _controller!.value.isPlaying
      ? WakelockPlus.enable()
      : WakelockPlus.disable();
  }
```

?